src/ghga_datasteward_kit/batch_s3_upload.py

```diff
@@ -213,6 +252,103 @@ def trigger_file_upload(self, file: FileMetadata) -> subprocess.Popen | None:
+
+        # Calculate days, hours, and minutes
+        days = int(seconds / SEC_IN_DAY)
+        remaining_seconds_after_days = seconds % SEC_IN_DAY
+        hours = int(remaining_seconds_after_days / SEC_IN_HOUR)
+        remaining_seconds_after_hours = remaining_seconds_after_days % SEC_IN_HOUR
+        minutes = int(remaining_seconds_after_hours / 60)
+        remaining_seconds_after_minutes = remaining_seconds_after_hours % 60
+
+        if remaining_seconds_after_minutes > 0 and minutes == 0:
+            # If there are remaining seconds but no full minutes, we set minutes to 1
+            minutes = 1
+        return f"{days:02}d {hours:02}h {minutes:02}m"
+
+    def _format_bytes(self, size_bytes: int) -> str:
+        """Formats bytes into a human-readable string (B, KB, MB, GB)."""
+        if not isinstance(size_bytes, int) or size_bytes < 0:
+            return "N/A"
+        if size_bytes == 0:
+            return "0 B"
+        suffixes = ["B", "KB", "MB", "GB", "TB", "PB"]
+        power = 0
+        # Determine the appropriate suffix by dividing by 1024
+        num = float(size_bytes)
+        while num >= 1024 and power < len(suffixes) - 1:
+            num /= 1024
+            power += 1
+        return f"{num:.2f} {suffixes[power]}"
+
+    def _calculate_eta(self, file: FileMetadata, recalc_speed: bool) -> str:
+        """Calculate the estimated time remaining for the batch.
+
+        ETA is updated after each completed file upload.
+        If `recalc_speed` is True, the upload rate is recalculated and stored.
+        If `recalc_speed` is False, the previously stored upload rate is used.
+        """
+        # Get time elapsed since the file upload started and remove it from tracking
+        file_size = self.file_sizes.get(file, 0)
+
+        # Update total bytes processed
+        self.total_bytes_processed += file_size
+
+        # If required, update the stored upload rate
+        if recalc_speed and self.total_bytes_processed > 0:
+            current_time = monotonic()
+            upload_duration = max(current_time - self.file_start_times.pop(file), 1)
+            # Calculate average rate of most recent upload in bytes per second
+            self.latest_upload_rate_bps = file_size / upload_duration
+
+        eta = "N/A"
+
+        # If the stored upload rate is available, estimate the remaining time
+        if self.latest_upload_rate_bps:
+            bytes_left = self.total_initial_bytes - self.total_bytes_processed
+            estimated_remaining_seconds = bytes_left / self.latest_upload_rate_bps
+            eta = self._format_time(estimated_remaining_seconds)
+
+        return eta
+
+    def _print_upload_progress(
+        self, file: FileMetadata, success: bool, skipped: bool = False
+    ):
+        """Print the upload progress for the batch upon finishing the given file."""
+        file_size = self.file_sizes.get(file, 0)
+        eta = self._calculate_eta(file, recalc_speed=success and not skipped)
```

Comment:
Why not update if it was unsuccessful?
If bytes were moved around during that time, then that influences the download speed and ETA.